flake-module.nix

```diff
@@ -0,0 +1,83 @@
+{ self, lib, flake-parts-lib, ... }:
+let
+  inherit (flake-parts-lib)
+    mkPerSystemOption;
+  inherit (lib)
+    mdDoc
+    mkOption
+    types;
+in
+{
+  options = {
+    perSystem = mkPerSystemOption
+      ({ config, self', inputs', pkgs, system, ... }: {
+        options.process-compose = mkOption {
+          description = mdDoc ''
+            process-compose-flake: creates process-compose executables
+            from your Nix attribute sets.
+          '';
+          type = types.submodule {
+            options = {
+              package = mkOption {
+                type = types.package;
+                default = pkgs.process-compose;
+              };
+              configs = mkOption {
+                type = types.attrsOf types.attrs;
+                default = { };
+                example = {
+                  # apps.${system}.watch-server and packages.${system}.watch-server become available
+                  # execute `nix run .#watch-server` or incude it as a buildInput to your devShell
+                  watch-server = {
+                    processes = {
+                      backend = "${pkgs.simple-http-server}";
+                      frontend = "${pkgs.simple-http-server}";
```

Comment:
```suggestion
                      backend.command = "${pkgs.simple-http-server}";
                      frontend.command = "${pkgs.simple-http-server}";
```

I think?

Reply:
Also the whole example should be in a string in `lib.literalExample`, escaping the `$`.

Reply:
Didn't know about `literalExample` thank you!

Reply:
@roberth `literalExample` is deprecated https://github.com/NixOS/nixpkgs/blob/2718b0829f065c722f53de77e329696b02da23c2/lib/options.nix#L290

Reply:
Ah, I meant `literalExample`. I think I might even have merged that deprecation, but muscle memory is a thing...